integrations/transportation/tfl.js

```diff
@@ -0,0 +1,173 @@
+import moment from 'moment';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PUBLIC_TRANSPORT } from '../../definitions';
+import { HTTPError, ValidationError } from '../utils/errors';
+
+const LOGIN_PATH = 'https://account.tfl.gov.uk/api/login';
+const BASE_PATH = 'https://mobileapi.tfl.gov.uk'; // For fetching data
+
+
+function generateActivities(travelDays) {
+  const activities = [];
+  travelDays.forEach((day) => {
+    day.Journeys.forEach((journey) => {
+      if (journey.Charge <= 0) { // ignore topups
+        activities.push({
+          id: journey.StartTime, // a string that uniquely represents this activity
+          datetime: journey.StartTime, // a javascript Date object that represents the start of the activity
+          durationHours: journey.EndTime && moment(journey.EndTime).diff(moment(journey.StartTime)) / (60 * 60 * 1000), // a floating point that represents the duration of the activity in decimal hours
+          distanceKilometers: journey.TransactionType === 'Bus' ? 3.140 : null, // a floating point that represents the amount of kilometers traveled (https://www.whatdotheyknow.com/request/bus_passenger_journey_times)
+          activityType: ACTIVITY_TYPE_TRANSPORTATION,
+          transportationMode: TRANSPORTATION_MODE_PUBLIC_TRANSPORT, // a variable (from definitions.js) that represents the transportation mode
+          carrier: 'Transport For London', // (optional) a string that represents the transportation company
+          departureStation: journey.TransactionType === 'Bus' ? 'Bus' : journey.StartLocation, // (for other travel types) a string that represents the original starting point
+          destinationStation: journey.TransactionType === 'Bus' ? 'Bus' : journey.EndLocation, // (for other travel types) a string that represents the final destination
+        });
+      }
+    });
+  });
+  return activities;
+}
+
+async function connect(requestLogin) {
+  const { username, password } = await requestLogin();
+  if (!(password || '').length) {
+    throw new ValidationError('Password cannot be empty');
+  }
+
+  const postBody = {
+    username,
+    password,
+    AppId: '9C9C6B6C-A025-493E-8F39-3A6D57C7ACAB', // TODO what do we do with this?
```

Comment:
Let's move that to env.json (you can look at the tripit integration for an example of how an environment variable is used)

Reply:
I'd also encourage a **rebase** to keep this key out of the history

Reply:
I've removed the appId, will rebase too...

Reply:
Where shall we store the appId?